app/src/main/java/com/reminder/hydration/EntryActivity.kt

```diff
@@ -0,0 +1,74 @@
+package com.reminder.hydration
+
+import android.annotation.SuppressLint
+import android.graphics.Color
+import android.os.Bundle
+import android.widget.*
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+
+class EntryActivity : AppCompatActivity() {
+
+    private val sharedPrefManager by lazy {
+        SharedPrefManager.getInstance(this)
+    }
+
+    @SuppressLint("WrongViewCast")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_entry)
+
+        val waterReminderNotifGroup = findViewById<RadioGroup>(R.id.waterReminderNotifGroup)
+        val notificationSoundGroup = findViewById<RadioGroup>(R.id.notificationSoundGroup)
+        val entryPageSubmitButton = findViewById<Button>(R.id.entryPageSubmitButton)
+        val entryPageMessage = findViewById<TextView>(R.id.entryPageMessage)
+
+        val viewModel: EntryViewModel by viewModels {
+            EntryViewModelFactory(sharedPrefManager)
+        }
+
+        val wakeUpTime = findViewById<EditText>(R.id.wakeUpTimeH)
+        val sleepTime = findViewById<EditText>(R.id.sleepTimeH)
+        val waterNum = findViewById<EditText>(R.id.drunkenWaterAmount)
+
+        viewModel.loadData()
+        viewModel.wakeUpTime().observe(this) {
+            wakeUpTime.setText(it.toString())
+        }
+        viewModel.sleepTime().observe(this) {
+            sleepTime.setText(it.toString())
+        }
+        viewModel.drunkenWaterAmount().observe(this) {
+            waterNum.setText(it.toString())
+        }
+        viewModel.notifType().observe(this) {
+            waterReminderNotifGroup.check(EnumToID.of(NotifType.of(it)))
+        }
+        viewModel.soundType().observe(this) {
+            notificationSoundGroup.check(EnumToID.of(SoundType.of(it)))
+        }
```

Comment:
Too many observers for the view. Remember we have 3 layers. Model, View and ViewModel. You are using View and ViewModel but where is the model in this? You are passing the raw data you have around. You should create a Model class which contains fields for each of this data and pass that one to the View and use it. A data class is perfect here.